### PR TITLE
Use vanilla Kubernetes manifests for Rancher (replacing Route et al.)

### PR DESCRIPTION
**Pull request by bittner**
The changes herein will allow to generate a [vanilla Kubernetes](https://kubernetes.io/) deployment configuration, one that is usable with a [Rancher cluster](https://rancher.com/) (as opposed to [OpenShift v3](https://www.openshift.com/), which was exclusively targeted / supported until now).

### CODE REVIEW

**Review by chamer81 — CHANGES_REQUESTED** [review]
The test failures are caused by the following problems:

The Bandit test fails due to a security issue, see: https://bandit.readthedocs.io/en/latest/plugins/b602_subprocess_popen_with_shell_equals_true.html
Essentially the "Shell" class (in post_gen_project.py) is vulnerable to shell injection. If we are OK with this risk, then we should add an exception for bandit to ignore this problem.

The Behave tests are failing on Travis CI because the tox environment is somehow not able to install Kustomize:
`When I generate the deployment configuration         # tests/acceptance/steps/when.py:28 0.223s

      Assertion Failed: Command failed with error:

      Binary for kustomize not found. Attempting download ...

      Downloading https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize/v3.8.2/kustomize_v3.8.2_linux_amd64.tar.gz ...

      Extracting from /tmp/tmpch27hkwikustomize_v3.8.2_linux_amd64.tar.gz: /home/travis/build/painless-software/painless-continuous-delivery/.tox/behave/local/bin/kustomize

      Extracting binary failed: file could not be opened successfully

      Try installing in --user space.`

The kustomize-wrapper is included in the kubernetes installdeps, but there seems to be some problem getting the actual binary. I'm not sure what the problem is -- the binary that is referenced in the above error message does exist. Maybe there's not enough space in the Travis CI runner to extract it....?

Anyway, I think we should clean up these test problems because when the test are showing as red, it's hard to identify new problems that may arise.

**Comment by bittner** [comment]
Argh, @chamer81, that's a new problem. and this is totally Travis CI specific. :worried: It could be that this is really a temporary Travis infrastructure issue BECAUSE the second Travis job actually _succeeds_. Very weird. I'll retrigger the failed build to see if this fixes the problem.

**Comment by bittner** [comment]
That fixed it. It was a Travis infrastructure hickup.

We now only need the Kubernetes manifests added to this PR, then we can merge it. @ZuGao please! :rocket:

**Comment by soufianebenali** [comment]
LGTM

**Review by soufianebenali — APPROVED** [review]

**Review by chamer81 — APPROVED** [review]
Looks good, thanks for fixing the tests!

**Comment by ZuGao** [comment]
Updated the manifests to work properly with springboot and flux

File: cookiecutter.json [hunk 1]
```diff
@@ -43,6 +44,10 @@
     ],
     "gitops_project": "{% if cookiecutter.deployment_strategy == 'gitops' %}{{ cookiecutter.project_slug }}-gitops{% else %}(none){% endif %}",
     "docker_registry": "{{ cookiecutter.cloud_platform.replace('APPUiO', 'registry.appuio.ch').replace('(none)', cookiecutter.vcs_platform.replace('Bitbucket.org', 'hub.docker.com').replace('GitHub.com', 'quay.io').replace('GitLab.com', 'registry.gitlab.com') + '/' + cookiecutter.vcs_account) }}",
+    "docker_image": "{{ cookiecutter.project_slug }}",
+    "registry_user": "{{ cookiecutter.ci_service.replace('.yml', '').replace('.', '').replace('-steps', '').replace('travis', 'travis-ci') }}",
+    "automation_user": "{{ cookiecutter.ci_service.replace('.yml', '').replace('.', '').replace('-steps', '').replace('travis', 'travis-ci') }}",
+    "flux_ssh_known_hosts": "",
```
Comment (bittner) [inline cookiecutter.json:50]: @ZuGao This is not in use yet. I'll remove it for this PR to make sense to be merged.

You can still add it later, but I'd suggest to use a more generic naming (something with "gitops" maybe) or have this added in some other way.

File: {{cookiecutter.project_slug}}/_/frameworks/SpringBoot/pom.xml [hunk 2]
```diff
@@ -36,6 +36,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-actuator</artifactId>
+        </dependency>
```
Comment (bittner) [inline {{cookiecutter.project_slug}}/_/frameworks/SpringBoot/pom.xml:42]: As explained by @ZuGao yesterday, this is a useful addition to the Spring Boot setup to easily provide health check endpoints for the Kubernetes deployment configuration.

Noteworthy to remember for all of us! :point_up:

File: {{cookiecutter.project_slug}}/deployment/application/base/deployment.yaml [hunk 3]
```diff
@@ -30,19 +33,31 @@ spec:
...
           initialDelaySeconds: 13
         readinessProbe:
+          {%- if cookiecutter.framework == 'SpringBoot' %}
+          httpGet:
+            path: /actuator/health/readiness
+            port: uwsgi
+          {%- else %}
           tcpSocket:
             port: uwsgi
+          {%- endif %}
```
Comment (bittner) [inline {{cookiecutter.project_slug}}/deployment/application/base/deployment.yaml:60]: Just as a heads-up: We need to refactor this part.

1. Health checks (probes) should be generic, implemented for all frameworks.
1. This has nothing to do with "uwsgi" specifically here (only with Python).

**Pull request merged**